src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2280,4 +2280,13 @@ private static String maybeQuoteETag(String eTag) {
     private static boolean startsWithIgnoreCase(String string, String prefix) {
         return string.toLowerCase().startsWith(prefix.toLowerCase());
     }
+
+    // Protects against illegal characters in blob names only if the client requests it
+    private String encodeBlob(String encodingType, String blobName) {
+        if (encodingType.equals("url")) {
```

Comment:
.equals in the function does that by default. Supposedly.

Sent from my iPhone

On Dec 11, 2015, at 9:25 PM, Andrew Gaul <notifications@github.com> wrote:

In src/main/java/org/gaul/s3proxy/S3ProxyHandler.java
<https://github.com/andrewgaul/s3proxy/pull/106#discussion_r47425453>:

> @@ -2280,4 +2280,13 @@ private static String maybeQuoteETag(String eTag) {
>      private static boolean startsWithIgnoreCase(String string, String prefix) {
>          return string.toLowerCase().startsWith(prefix.toLowerCase());
>      }
> +
> +    // Protects against illegal characters in blob names only if the client requests it
> +    private String encodeBlob(String encodingType, String blobName) {
> +        if (encodingType.equals("url")) {

HttpServletRequest.getParameter can return null and thus we must check
encodingType here.

—
Reply to this email directly or view it on GitHub
<https://github.com/andrewgaul/s3proxy/pull/106/files#r47425453>.